### PR TITLE
Add UI for paid domains

### LINKED ISSUE #17899: Update free domain suggestions UI
The UI of free domain suggestions will be updated.
The view is displayed after the `Choose a theme` screen, when the search input is not empty.

#### Subtasks

- [ ] Update free domain list item UI

### Design

| Before | After |
| - | - |
|  |  |

---
> **Note**:
> - Other changes are out of scope and part of other tasks
> - All development should be behind [the feature flag](https://github.com/wordpress-mobile/WordPress-Android/issues/17896)

### LINKED ISSUE #17985: Add UI for paid domains
Part of #17900

#### Subtasks

- [ ] Display the cost as `cost/yr`
- [ ] Add dot indicator and the following labels:
  - 1st item: `Recommended`
  - 2nd item: `Best alternative`

### Design

---
> **Note**:
> - Other changes are out of scope and part of other tasks
> - All development should be behind [the feature flag](https://github.com/wordpress-mobile/WordPress-Android/issues/17896)

### PR DESCRIPTION
**Pull request by ovitrif**
Resolves #17985

This PR mainly implements the UI for paid domains:
- ★ Implements domain item variants and logic: 1st: `Recommended`, 2nd: `Best Alternative`
- ⊕ Adds UI for showing first item as unavailable — implementation to be done in other task
- ⊕ Adds UI for domains on sale — implementation to be done in other task

#### Prerequisite

<details>
<summary>◐ Toggle the <code>SiteCreationDomainPurchasingFeatureConfig</code> flag</summary>

  1. Go to `Me` → `App Settings` → `Debug settings`
  2. Scroll to the `Features in development` section
  3. Tap on `SiteCreationDomainPurchasingFeatureConfig`
  4. Tap `RESTART THE APP` button

  More detailed steps where to verify:
  - https://github.com/wordpress-mobile/WordPress-Android/pull/17998
  ---
</details>

### ● Treatment Variation

- **Verify** 1st domain suggestion shows as `Recommended`
- **Verify** 2nd domain suggestion shows as `Best Alternative`
- **Verify** domain cost is shown as designed

### ○ Control Variation

- **Verify** No UI changes in `Site creation` → `Domains`

### Previews

paidUI

## Regression Notes

1. Potential unintended areas of impact
   `Site creation` → `Domains`

2. What I did to test those areas of impact (or what existing automated tests I relied on)
   Manual testing and unit testing

3. What automated tests I added (or what prevented me from doing so)
   Unit tests in `SiteCreationDomainsViewModelTest`

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by ovitrif** [comment]
Closed in favor of #17998

File: WordPress/src/main/java/org/wordpress/android/ui/compose/components/SolidCircle.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.Dp
+
+@Composable
+fun SolidCircle(
```
Comment (antonis) [inline WordPress/src/main/java/org/wordpress/android/ui/compose/components/SolidCircle.kt:14]: Thank you for adding this reusable component 👍

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/domains/compose/DomainItem.kt [hunk 2]
```diff
@@ -2,57 +2,151 @@ package org.wordpress.android.ui.sitecreation.domains.compose
...
-            Text(
-                text = cost,
-                color = CostColorDefault,
-            )
+            Box(
+                contentAlignment = Alignment.Center,
+                modifier = Modifier.width(Margin.ExtraMediumLarge.value)
+            ) {
+                variant?.dotColor?.let {
+                    SolidCircle(size = 8.dp, colorResource(it))
```
Comment (antonis) [inline WordPress/src/main/java/org/wordpress/android/ui/sitecreation/domains/compose/DomainItem.kt:54]: My understanding is that the reasoning for inlining dimension values is [similar with the previous PR](https://github.com/wordpress-mobile/WordPress-Android/pull/17998#discussion_r1114999461). Correct?

**Review by antonis — APPROVED** [review]
Great work @ovitrif 👍 
I've tested the UI implementation on a Pixel 5 with Android 13 and everything worked as described for me. The code also looks great 🎉

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/domains/compose/DomainItem.kt (hunk 2 above)
Comment (ovitrif) [reply WordPress/src/main/java/org/wordpress/android/ui/sitecreation/domains/compose/DomainItem.kt:54]: Correct 🙇🏻 , we're (Ajesh and I) probably going do start a discussion (internally) on this very soon 👍🏻

File: WordPress/src/main/java/org/wordpress/android/ui/compose/components/SolidCircle.kt (hunk 1 above)
Comment (ovitrif) [reply WordPress/src/main/java/org/wordpress/android/ui/compose/components/SolidCircle.kt:14]: > Thank you for adding this reusable component 👍

Thanks @antonis 🙇🏻 
For naming I took inspiration from SwiftUI which calls it `Circle`. @AjeshRPai  was also interested in this component to potentially replace a [different solution](https://github.com/wordpress-mobile/WordPress-Android/pull/17941/files#diff-b85e2edbb71dce7062cc0c64df4bd322419a4379c191fe00b66dd42ca9fed120R258-R263) added recently for Blaze 👍🏻

**Pull request merged**